safe-global/web-core
# feat: mobile pairing

## What it solves

Adding mobile pairing

## How this PR fixes it

A WalletConnect connector is initialsed outside of the app and listened to by `useInitPairing` for a handful of events to manage the session (dis-)connection.

After successful connection, the connector is linked to onboard and works as a normal module.

## How to test it

Scanning the QR code from the wallet popper or the Safe creation flow should connect as expected, allowing for interaction with the Safe.

It is not possible to open the wallet selection and choose "Safe Mobile". A new QR modal should open, allowing for scanning as above, connecting as a normal wallet.

Bugs will likely be found regarding multipl variations of (dis-)connection:

- Connecting via scanning then disconnecting via the app or via the UI. Chaining these methods together may showcase bugs.
- Switching between chains (whilst not connected) should adjust the session to match the new chain, but this could also be buggy.

## Analytics changes

Clicking the "App Store" button will track a new event with our campaign link. Connecting the pairing module should also be tracked as a wallet like others are.

## Screenshots

![image](https://user-images.githubusercontent.com/20442784/187188166-39a8b2e7-588e-4334-a061-b5ab665383a6.png)

![image](https://user-images.githubusercontent.com/20442784/187188241-7362628b-df9e-4641-9379-63c296022659.png)

![image](https://user-images.githubusercontent.com/20442784/187188219-d84ddf40-a9fa-49c1-a308-72407972a79a.png)

![image](https://user-images.githubusercontent.com/20442784/187188192-90d2b0a2-ab77-4d19-ac66-8d4eabdfdb50.png)

![image](https://user-images.githubusercontent.com/20442784/187188331-e71ec5f5-684e-43c6-b437-9800ef58cc04.png)
